OpenRA.Game/Traits/World/MusicPlaylist.cs

```diff
@@ -41,44 +36,57 @@ public class MusicPlaylistInfo : ITraitInfo
 	public class MusicPlaylist : INotifyActorDisposing, IGameOver
 	{
 		readonly MusicPlaylistInfo info;
+		readonly World world;
 
 		readonly MusicInfo[] random;
 		readonly MusicInfo[] playlist;
 
 		public readonly bool IsMusicAvailable;
+		public bool CurrentSongIsBackground { get; private set; }
 
 		MusicInfo currentSong;
-		bool repeat;
+		MusicInfo currentBackgroundSong;
 
 		public MusicPlaylist(World world, MusicPlaylistInfo info)
 		{
 			this.info = info;
-
-			IsMusicAvailable = world.Map.Rules.InstalledMusic.Any();
+			this.world = world;
 
 			playlist = world.Map.Rules.InstalledMusic.Select(a => a.Value).ToArray();
 
+			IsMusicAvailable = playlist.Length > 1;
```

Comment:
This was inherited from @obrakmann's original implementation.  I've removed it for now so we can argue about it in a followup PR.  See #8747.